contrib/kubespray/script/validate_layout_schema.py

```diff
@@ -0,0 +1,118 @@
+import argparse
+import logging
+import logging.config
+import sys
+from schema import Schema, Or, Optional, Regex
+import yaml
+
+
+def setup_logger_config(logger):
+    """
+    Setup logging configuration.
+    """
+    if len(logger.handlers) == 0:
+        logger.propagate = False
+        logger.setLevel(logging.DEBUG)
+        consoleHandler = logging.StreamHandler()
+        consoleHandler.setLevel(logging.DEBUG)
+        formatter = logging.Formatter('%(asctime)s [%(levelname)s] - %(filename)s:%(lineno)s : %(message)s')
+        consoleHandler.setFormatter(formatter)
+        logger.addHandler(consoleHandler)
+
+logger = logging.getLogger(__name__)
+setup_logger_config(logger)
+
+
+def load_yaml_config(config_path):
+    with open(config_path, "r") as f:
+        config_data = yaml.load(f, yaml.SafeLoader)
+    return config_data
+
+
+def validate_layout_schema(layout):
+    schema = Schema(
+        {
+            'machine-sku': {
+                str: {
+                    'mem': str,
+                    'cpu': {
+                        'vcore': int,
+                    },
+                    Optional('computing-device'): {
+                        'type': str,
+                        'model': str,
+                        'count': int,
+                    }
+                }
+            },
+            'machine-list': [
+                {
+                    'hostname': str,
```

Comment:
For hostname, please use this regex: `[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`

See https://github.com/kubernetes-sigs/kubespray/blob/release-2.11/roles/kubernetes/preinstall/tasks/0020-verify-settings.yml#L124

Reply:
> For hostname, please use this regex: `[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
> 
> See https://github.com/kubernetes-sigs/kubespray/blob/release-2.11/roles/kubernetes/preinstall/tasks/0020-verify-settings.yml#L124

Fixed, thx